spec/models/admin_dashboard_data_spec.rb

```diff
@@ -3,37 +3,129 @@
+        expect(problems).not_to include(I18n.t("errors.messages.invalid"))
+      end
+    end
+
+    describe "adding new checks" do
+      it 'calls the passed block' do
+        called = false
+        AdminDashboardData.add_problem_check do
+          called = true
+        end
+
+        AdminDashboardData.fetch_problems
+        expect(called).to eq(true)
+
+        AdminDashboardData.fetch_problems(check_force_https: true)
+        expect(called).to eq(true)
+      end
+
+      it 'calls the passed method' do
+        $test_AdminDashboardData_global = false
+        class AdminDashboardData
+          def my_test_method
+            $test_AdminDashboardData_global = true
+          end
+        end
+        AdminDashboardData.add_problem_check :my_test_method
 
-  describe "adding new checks" do
-    after do
-      AdminDashboardData.reset_problem_checks
+        AdminDashboardData.fetch_problems
+        expect($test_AdminDashboardData_global).to eq(true)
+        $test_AdminDashboardData_global = nil
+      end
     end
+  end
 
-    it 'calls the passed block' do
+  describe "adding scheduled checks" do
+    it "adds the passed block to the scheduled checks" do
       called = false
-      AdminDashboardData.add_problem_check do
+      AdminDashboardData.add_scheduled_problem_check(:test_identifier) do
         called = true
       end
 
-      AdminDashboardData.fetch_problems
-      expect(called).to eq(true)
-
-      AdminDashboardData.fetch_problems(check_force_https: true)
+      AdminDashboardData.execute_scheduled_checks
       expect(called).to eq(true)
     end
 
-    it 'calls the passed method' do
-      $test_AdminDashboardData_global = false
-      class AdminDashboardData
-        def my_test_method
-          $test_AdminDashboardData_global = true
-        end
+    it "adds a found problem from a scheduled check" do
+      AdminDashboardData.add_scheduled_problem_check(:test_identifier) do
+        AdminDashboardData::Problem.maybe_create("test problem")
       end
-      AdminDashboardData.add_problem_check :my_test_method
 
-      AdminDashboardData.fetch_problems
-      expect($test_AdminDashboardData_global).to eq(true)
-      $test_AdminDashboardData_global = nil
+      AdminDashboardData.execute_scheduled_checks
+      problems = AdminDashboardData.load_found_scheduled_check_problems
+      expect(problems.first).to be_a(AdminDashboardData::Problem)
+      expect(problems.first.message).to eq("test problem")
+    end
+
+    it "does not add duplicate problems with the same identifier" do
+      prob1 = AdminDashboardData::Problem.maybe_create("test problem", identifier: "test")
+      prob2 = AdminDashboardData::Problem.maybe_create("test problem 2", identifier: "test")
+      AdminDashboardData.add_found_scheduled_check_problem(prob1)
+      AdminDashboardData.add_found_scheduled_check_problem(prob2)
+      expect(AdminDashboardData.load_found_scheduled_check_problems.count).to eq(1)
```

Comment:
Hehe I have the same concern as the above. It isn't clear from the assertion which problem we expect to return here.